### PR TITLE
display more revert information in audit log

### PR DESCRIPTION
**Pull request by Bliztle**
Involved refactoring commit store to allow for proper reverts. Last structure didnt work with non-parameter-reverts

### CODE REVIEW

File: pmp-client/src/features/audit/AuditTableRow.tsx [hunk 1]
```diff
@@ -4,11 +4,31 @@ import AuditDetailsDialog from './AuditDetailsDialog';
...
+//     };
+
+//     return (
+//         <>
+//             <Tooltip content='Revert commit'>
+//                 <IconButton icon={{ icon: 'undo' }} onClick={handleClick} />
+//             </Tooltip>
+//         </>
+//     );
+// };
```
Comment (TheMagzuz) [inline pmp-client/src/features/audit/AuditTableRow.tsx:25]: Commented code block

File: pmp-client/src/features/audit/AuditTableRow.tsx [hunk 2]
```diff
@@ -18,6 +38,9 @@ const AuditTableRow = ({ entry }: AuditRowProps) => {
                 <DataTableCell>{entry.email}</DataTableCell>
                 <DataTableCell>{entry.hash}</DataTableCell>
                 <DataTableCell>{entry.message}</DataTableCell>
+                {/* <DataTableCell>
+                    <AuditTableRowRevert entry={entry} />
+                </DataTableCell> */}
```
Comment (TheMagzuz) [inline pmp-client/src/features/audit/AuditTableRow.tsx:43]: Commented

File: pmp-client/src/features/audit/connecting mock commits.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
+# Connecting mock commits
```
Comment (TheMagzuz) [inline pmp-client/src/features/audit/connecting mock commits.md:1]: What is the purpose of this file?

File: pmp-client/src/features/changes/commitStore.ts [hunk 4]
```diff
@@ -55,81 +54,57 @@ export const createCommitStore = (storageKey: string) => {
...
+                    const change = get()
+                        .changes.filter(isParameterChange)
+                        .find((c) => {
+                            if (c.service.name !== service.name) {
+                                return false;
+                            }
+                            if (c.parameter.name !== parameter.name) {
+                                return false;
+                            }
+                            return true;
```
Comment (TheMagzuz) [inline pmp-client/src/features/changes/commitStore.ts:67]: Could this be simpler by just returning the comparison?

File: pmp-client/src/features/changes/commitStoreHelpers.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    if (isParameterChange(a) && isParameterChange(b)) {
+        return compareParameterChanges(a, b);
+    }
+
+    if (isRevert(a)) {
+        return -1;
+    }
+
+    return 1;
```
Comment (TheMagzuz) [inline pmp-client/src/features/changes/commitStoreHelpers.ts:28]: Why are these different?

File: pmp-client/src/features/changes/commitStoreHelpers.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    const typeComparison = a.revertType.localeCompare(b.revertType);
+    if (typeComparison !== 0) return typeComparison;
+
+    return a.commitReference.localeCompare(b.commitReference);
+};
+
+export const areRevertsEqual = (a: Revert, b: Revert): boolean => {
+    if (isParameterRevert(a) && isParameterRevert(b)) {
+        return a.parameterName === b.parameterName;
+    }
```
Comment (TheMagzuz) [inline pmp-client/src/features/changes/commitStoreHelpers.ts:48]: Are two different parameter reverts that revert the same parameter to different times not different?

File: pmp-client/src/features/changes/commitStoreHelpers.ts (hunk 5 above)
Comment (Bliztle) [reply pmp-client/src/features/changes/commitStoreHelpers.ts:28]: a is revert, b is not => should be ordered one way
b is revert, a is not => should be ordered the other way

**Pull request closed without merging**